src/unix/haiku/native.rs

```diff
@@ -197,6 +197,50 @@ e! {
         B_UTILITIES_DIRECTORY,
         B_PACKAGE_LINKS_DIRECTORY,
     }
+
+    // kernel/OS.h
+
+    pub enum topology_level_type {
+        B_TOPOLOGY_UNKNOWN,
+        B_TOPOLOGY_ROOT,
+        B_TOPOLOGY_SMT,
+        B_TOPOLOGY_CORE,
+        B_TOPOLOGY_PACKAGE,
+    }
```

Comment:
Is this valid? enums in Rust and C have very different semantics -- in C they are integer types and some associated constants, in Rust a value of enum type must *always* be one of the variants or else there is immediate UB.

The `e!` macro does not even seem to add any `repr` so the layout of these enums seems to be largely undefined as well.